backend/LocalWebApp/BackgroundSyncService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Threading.Channels;
+using Crdt;
+using LcmCrdt;
+using MiniLcm;
+
+namespace LocalWebApp;
+
+public class BackgroundSyncService(
+    IServiceProvider serviceProvider,
+    ProjectsService projectsService,
+    ProjectContext projectContext) : BackgroundService
+{
+    private readonly Channel<CrdtProject> _syncResultsChannel = Channel.CreateUnbounded<CrdtProject>();
+
+    public void TriggerSync()
+    {
+        _syncResultsChannel.Writer.TryWrite(projectContext.Project ??
+                                            throw new InvalidOperationException("No project selected"));
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        await Task.Delay(1_000, stoppingToken);
+        var crdtProjects = await projectsService.ListProjects();
+        foreach (var crdtProject in crdtProjects)
+        {
+            await SyncProject(crdtProject);
+        }
+
+        if (!projectsService.ProjectExists("sena-3"))
+        {
+            await projectsService.CreateProject("sena-3",
+                new Guid("0ebc5976-058d-4447-aaa7-297f8569f968"), //same as sena 3 project id in SeedDatabase
+                "http://localhost:5158",
+                async (provider, project) =>
+                {
+                    var (_, _, isSynced) = await SyncProject(project);
+                    //seed will only create if missing, so seed anyway since the project should exist on the server always
+                    await SeedDb(provider.GetRequiredService<ILexboxApi>());
+                });
+        }
+
+        await foreach (var project in _syncResultsChannel.Reader.ReadAllAsync(stoppingToken))
```

Comment:
It looks to me like reading only happens once on startup, but that writing only ever happens afterwards, so this doesn't do anything at the moment right?

Reply:
this waits here, until something gets written, it only finishes looping if `_syncResultsChannel.Writer.Complete();` is called.